crates/sozo/src/commands/options/mod.rs

```diff
@@ -1,10 +1,31 @@
-use scarb::core::Workspace;
+use scarb::core::{ManifestMetadata, Workspace};
 use toml::Value;
 
 pub mod account;
 pub mod starknet;
 pub mod world;
 
-pub(super) fn dojo_metadata_from_workspace(ws: &Workspace<'_>) -> Option<Value> {
-    ws.current_package().ok()?.manifest.metadata.tool_metadata.as_ref()?.get("dojo").cloned()
+pub(crate) fn dojo_metadata_from_workspace(ws: &Workspace<'_>) -> Option<DojoMetadata> {
+    Some(ws.current_package().ok()?.manifest.metadata.dojo_metadata())
+}
+
+pub(crate) struct DojoMetadata {
+    env: Option<Value>,
+}
+
+impl DojoMetadata {
+    pub fn env(&self) -> Option<Value> {
```

Comment:
```suggestion
    pub fn env(&self) -> Option<&Environment> {
```

refer to earlier comments